swift/Sources/PiranhaKit/InputCommand/PiranhaConfigProvider.swift

```diff
@@ -0,0 +1,35 @@
+/**
+ *    Copyright (c) 2019 Uber Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+import ArgumentParser
+import Foundation
+
+protocol PiranhaConfigProviding {
+    func config(fromFileURL url: URL) throws -> PiranhaConfig
+}
+
+struct PiranhaConfigProviderDefaultImpl: PiranhaConfigProviding {
+    
+    func config(fromFileURL url: URL) throws -> PiranhaConfig {
+        do {
+            let properties = try Data(contentsOf: url)
+            return try JSONDecoder().decode(PiranhaConfig.self,
```

Comment:
Can we also make sure that the parsed JSON has flagname, treated or control, and flagIndex as required fields?

Reply:
Yeah, since we are now making use of Codable conformance and having required properties - (flagname, treated or control, and flagIndex) - the JSONDecoder will make sure that these properties are always present.